.github/workflows/python_package.yml

```
@@ -112,26 +141,22 @@ jobs:
       - name: Setup and run tests
         shell: bash
         run: |
+          export COMPILER="${{ matrix.compiler }}"
           export TASK="${{ matrix.task }}"
           export METHOD="${{ matrix.method }}"
           export OS_NAME="macos"
           export PYTHON_VERSION="${{ matrix.python_version }}"
-          if [[ "${{ matrix.os }}" == "macos-14" ]]; then
-              # use clang when creating macOS release artifacts
-              export COMPILER="clang"
-          else
-              export COMPILER="gcc"
-          fi
           export BUILD_DIRECTORY="$GITHUB_WORKSPACE"
           export CONDA=${HOME}/miniforge
           export PATH=${CONDA}/bin:${PATH}
+          export PRODUCES_ARTIFACTS="${{ matrix.produces-artifacts }}"
           $GITHUB_WORKSPACE/.ci/setup.sh || exit 1
           $GITHUB_WORKSPACE/.ci/test.sh || exit 1
       - name: upload wheels
```

For the consistency.

```suggestion
      - name: Upload artifacts
```